### DISCUSSION TITLE
Individual service instance by '/workspace:id'

### DISCUSSION THREAD

**Discussion by funkynine** [socketio/socket.io#4497] [Q&A]
I need to create someone service instance related to one room and then handle the logic for this room in the service 
How can I implement this with your lib? 

I was thinking your library gives a possibility to create individual socket (something like in react-router when we define route we can add a dynamic parameter like id)

My fast solution for the current issue created some const that contained 
const instances = new Map([
  ['roomId', new ServiceInstanceRelatedRoom()],
  etc ....
]);

But I think it is a terrible solution 

Could you please hint to me on how I should Implement it?

**Comment by darrachequesne** [comment]
Hi! You could use a dynamic namespace: https://socket.io/docs/v4/namespaces/#dynamic-namespaces
